bus-mapping/src/eth_types.rs

```diff
@@ -0,0 +1,485 @@
+                .unwrap(),
+        ),
+        to: Some(Address::zero()),
+        value: Word::from([0u8; 32]),
+        gas_price: Word::from([0u8; 32]),
+        gas: Word::from(1_000_000u64),
+        input: Bytes(Vec::new()),
+        v: Some(U64([0u64])),
+        r: Some(Word::from([0u8; 32])),
+        s: Some(Word::from([0u8; 32])),
+        raw: Some(Bytes(Vec::new())),
+        transaction_type: Some(U64([0u64])),
+        access_list: None,
+    }
+}
+
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize)]
+#[doc(hidden)]
+struct GethExecStepInternal {
+    pc: ProgramCounter,
+    op: OpcodeId,
+    gas: Gas,
+    #[serde(alias = "gasCost")]
+    gas_cost: GasCost,
+    depth: u8,
+    // stack is in hex 0x prefixed
+    stack: Vec<DebugU256>,
+    // memory is in chunks of 32 bytes, in hex
+    #[serde(default)]
+    memory: Vec<DebugU256>,
+    // storage is hex -> hex
+    #[serde(default)]
+    storage: HashMap<DebugU256, DebugU256>,
+}
+
+/// The execution step type returned by geth RPC debug_trace* methods.   Corresponds to
+/// `StructLogRes` in `go-ethereum/internal/ethapi/api.go`.
+#[derive(Clone, Debug, Eq, PartialEq)]
+#[doc(hidden)]
+pub struct GethExecStep {
+    pub pc: ProgramCounter,
+    pub op: OpcodeId,
+    pub gas: Gas,
+    pub gas_cost: GasCost,
+    pub depth: u8,
+    // stack is in hex 0x prefixed
+    pub stack: Stack,
+    // memory is in chunks of 32 bytes, in hex
+    pub memory: Memory,
+    // storage is hex -> hex
+    pub storage: Storage,
+}
+
+impl<'de> Deserialize<'de> for GethExecStep {
+    fn deserialize<D>(deserializer: D) -> Result<GethExecStep, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        let s = GethExecStepInternal::deserialize(deserializer)?;
+        Ok(GethExecStep {
+            pc: s.pc,
+            op: s.op,
+            gas: s.gas,
+            gas_cost: s.gas_cost,
+            depth: s.depth,
+            stack: Stack(
+                s.stack.iter().map(|dw| dw.to_word()).collect::<Vec<Word>>(),
+            ),
+            memory: Memory::from(
+                s.memory
+                    .iter()
+                    .map(|dw| dw.to_word())
+                    .collect::<Vec<Word>>(),
+            ),
+            storage: Storage(
+                s.storage
+                    .iter()
+                    .map(|(k, v)| (k.to_word(), v.to_word()))
+                    .collect(),
+            ),
+        })
+    }
+}
+
+/// The execution trace type returned by geth RPC debug_trace* methods.  Corresponds to
+/// `ExecutionResult` in `go-ethereum/internal/ethapi/api.go`.
+#[derive(Clone, Debug, Eq, PartialEq, Deserialize)]
+#[doc(hidden)]
+pub struct GethExecTrace {
+    pub gas: Gas,
+    pub failed: bool,
+    // return_value is a hex encoded byte array
+    #[serde(alias = "structLogs")]
+    pub struct_logs: Vec<GethExecStep>,
+}
+
+// TODO: Move this test macros to a crate, export them, and use them in all tests
```

Comment:
Can we file an issue and remove the TODO? On that way we will not forget about it :)

Reply:
done https://github.com/appliedzkp/zkevm-circuits/issues/164